app/assets/scripts/map.js

```diff
@@ -150,59 +159,76 @@ function create_marker(resource){
         $("#map").show();
         $("#resource-info").hide();
 
-        var contentDiv = document.createElement('div');
-        var resName = document.createElement('strong');
-        $(resName).html(resource.name);
-        var br = document.createElement('br');
-
-        // information pane to replace map
-        // TODO: refactor how the page elements are generated
-        var moreLink = document.createElement('p');
-        moreLink.innerHTML = 'More Info';
-        moreLink.setAttribute('class', 'more-info')
-        moreLink.addEventListener('click', function() {
+        // Show marker info bubble for clicked marker
+        var markerTemplate = $("#marker-template").html();
+        var compiledMarkerTemplate = Handlebars.compile(markerTemplate);
+        var context = {
+          name: resource.name,
+          address: resource.address,
+        };
+        var markerInfo = compiledMarkerTemplate(context);
+
+        if (infowindow) {
+          infowindow.close();
+        }
+        infowindow = new google.maps.InfoWindow({
+          content: markerInfo
+        });
+        infowindow.open(map, markerToAdd);
+        map.setCenter(markerToAdd.getPosition());
+        map.setZoom(focusZoom);
+
+        // Marker to detailed resource information view
+        $(".more-info").click(function() {
```

Comment:
I am not super well-versed in javascript style, but perhaps a large anonymous function within another function would be better off defined outside of the function?